roboflow/core/project.py

```diff
@@ -270,7 +271,10 @@ def upload(
             )
 
         if is_file:
-            is_image = self.check_valid_image(image_path) or is_hosted
+            if is_hosted:
```

Comment:
hit bug here that check_valid_image returns error when checking if `is_hosted`

Reply:
Yeah hmm - I'm going to go with:

* take is_hosted
* check the extension of image_path

The download and check maybe not a huge deal, but seems to defeat purpose of having everything happen up in the cloud - ideally infer server will handle good error there :D